### PR TITLE
[Merged by Bors] - chore(SetTheory/Cardinal/Basic): deprecate duplicate `Nat.cast` lemmas

### PR DESCRIPTION
**Pull request by vihdzp**
Spotted thanks to #17827.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 1]
```diff
@@ -1380,42 +1383,33 @@ theorem mk_finset_of_fintype [Fintype α] : #(Finset α) = 2 ^ Fintype.card α :
...
+theorem natCast_le {m n : ℕ} : (m : Cardinal) ≤ n ↔ m ≤ n := Nat.cast_le
 
--- porting note (#10618): simp can prove this
--- @[simp, norm_cast]
-@[norm_cast]
-theorem natCast_lt {m n : ℕ} : (m : Cardinal) < n ↔ m < n := by
-  rw [lt_iff_le_not_le, ← not_le]
-  simp only [natCast_le, not_le, and_iff_right_iff_imp]
-  exact fun h ↦ le_of_lt h
+@[deprecated Nat.cast_le (since := "2024-10-16")]
```
Comment (grunweg) [inline Mathlib/SetTheory/Cardinal/Basic.lean:1394]: This is a typo, right?
```suggestion
@[deprecated Nat.cast_lt (since := "2024-10-16")]

**Review by grunweg — APPROVED** [review]
LGTM so far; thanks for cleaning this up!

**Comment by grunweg** [comment]
Feel free to ping me when CI passes, and I can `maintainer merge` it.

**Comment by vihdzp** [comment]
@grunweg builds now!

**Review by grunweg — COMMENTED** [review]
Thanks!
maintainer merge

**Review by fpvandoorn — COMMENTED** [review]
bors merge

**Comment by fpvandoorn** [comment]
bors d+

**Pull request closed without merging**